shanoir-ng-datasets/src/main/java/org/shanoir/ng/dicom/DicomProcessing.java

```diff
@@ -26,18 +26,18 @@
 @Service
 public class DicomProcessing {
 	
-	public Attributes getDicomObjectAttributes(DatasetFile image, boolean isEnhancedMR) throws IOException {
+	public Attributes getDicomObjectAttributes(DatasetFile image, Boolean isEnhancedMR) throws IOException {
 		File dicomFile = new File(image.getPath());
 		try (DicomInputStream dIS = new DicomInputStream(dicomFile)) {
 			Attributes datasetAttributes;
-			if (isEnhancedMR) {
+			if (isEnhancedMR != null && isEnhancedMR) {
 				// In case of Enhanced MR, we need to the pixel data in order to use Dcm4chee emf extract method.
-				datasetAttributes = dIS.readDataset(-1, -1);
+				datasetAttributes = dIS.readDataset(-1);
 			} else {
 				// Else we do not load the picture in Ram for faster performance.
-				datasetAttributes = dIS.readDataset(-1, Tag.PixelData);
+				datasetAttributes = dIS.readDataset(Tag.PixelData);
```

Comment:
I'm not convinced this will work the same -> The second argument is a "stop to this tag" argument.
I think you can use this method: readDatasetUntilPixelData
(if existing on our version)

Reply:
from the source code : 

```
    /**
     * @deprecated Use one of the other {@link #readDataset()} methods instead. If you want to
     * specify a length limit, you may supply a {@link LimitedInputStream} or use
     * {@link #createWithLimit} or {@link #createWithLimitFromFileLength}.
     */
    @Deprecated
    public Attributes readDataset(int len, int stopTag) throws IOException {
        return readDataset(len, tagEqualOrGreater(stopTag));
    }

    public Attributes readDataset(int stopTag) throws IOException {
        return readDataset(tagEqualOrGreater(stopTag));
    }

    public Attributes readDataset(Predicate<DicomInputStream> stopPredicate) throws IOException {
        return readDataset(UNDEFINED_LENGTH, stopPredicate);
    }

    ...

    private static final int UNDEFINED_LENGTH = -1;
```